src/main/java/com/google/cloud/genomics/utils/Contig.java

```diff
@@ -99,7 +99,7 @@ public Contig apply(String contigString) {
   // the ShardUtils methods should be used to ensure that shards are shuffled all together before
   // being returned to clients.
   List<Contig> getShards(long numberOfBasesPerShard) {
-    double shardCount = Math.ceil(end - start) / numberOfBasesPerShard;
+    double shardCount = (end - start) / (double) numberOfBasesPerShard;
```

Comment:
Math.floor isn't what we want here.  See also the unit tests.

The useless Math.ceil there before caused a double cast on the numerator, which is why the old version "worked".